### DISCUSSION TITLE
How to configure .vitest/ path?

### DISCUSSION THREAD

**Discussion by zhao-li** [vitest-dev/vitest#1840] [Q&A] [answered]
Hi 👋,

Is there a way to configure Vitest to output to a location other than `node_modules/.vitest/`?

My container uses a named volume for `node_modules/` folder so it does not have permissions to modify that location:
```
⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Unhandled Error ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
Error: EACCES: permission denied, mkdir '/usr/src/app/node_modules/.vitest'
```

I've tried `outputFile` option: https://vitest.dev/config/#outputfile
But I don't think it's the one I am looking for because there doesn't seem to be an effect:
```
export default defineConfig({
  plugins: [vue()],
  resolve: {
    alias: {
      '@': fileURLToPath(new URL('./src', import.meta.url))
    },
  },
  test: {
    globals: true,
    environment: 'jsdom',
    outputFile: './artifacts/',
  },
})
```

Thank you for your time 🙏

**Comment by sheremet-va** [comment, marked as answer]
Use `test.cache.dir`

**Reply by zhao-li** [reply]
Thanks @sheremet-va ! that did it!

Would you also happen to know if there's a configuration that I can put inside `vite.config.js` that allows me to not need `--coverage` when running `vitest`?

Right now I tried to add this configuration:
```
export default defineConfig({
  ...
  test: {
    ...
    coverage: {
      statements: 100,
      branches: 100,
      functions: 100,
      lines: 100,
    },
  },
})
```
But I still need to execute `vitest --coverage`.

I want to be able to just run `vitest` and have the coverage flag enabled from `vite.config.js`.

Thank you again for your help and time 🙏
